lib/utility/runJoin.js

```diff
@@ -127,12 +129,13 @@ exports.runJoin = function runJoin(Model, isCollection, instances, models, isInn
 	 * @return limit the number of items in the query result
 	 */
 	function calculateLimit(model) {
+		var limit = 1;
 		if (model.multiple) {
-			//Limit rule for joining with Multiple Children
-			return (model.limit && (0 < model.limit && model.limit < 1001)) ? model.limit : 10;
-		} else {
-			return 1;
+			// since multiple is set then we want to query for more than one instance of the model.
+			limit = (model.limit && (0 < model.limit && model.limit < 1001)) ? model.limit : 10;
 		}
+		self.logger.debug('Set query limit to', limit);
```

Comment:
The idea was that from the logs we don't have an idea what the query was. I can remove if we really don't want.
The idea with these debug logs is that they are lower than trace. I was sure that the logs in prod are set to trace so you're just printing out these if running in -l debug locally. Could be totally wrong though....